sdk/src/packet.rs

```diff
@@ -37,30 +85,63 @@ pub struct Packet {
     pub meta: Meta,
 }
 
-impl Packet {
-    pub fn new(data: [u8; PACKET_DATA_SIZE], meta: Meta) -> Self {
-        Self { data, meta }
+// TODO: can we de-duplicate some of this Packet and ExtendedPacket code?
+#[derive(Clone)]
+#[repr(C)]
+pub struct ExtendedPacket {
+    pub data: [u8; EXTENDED_PACKET_DATA_SIZE],
```

Comment:
What if packet data was just a slice into a larger chunk of data stored in a "PacketBatch"? That way we don't need multiple types for packets

Reply:
Maybe; it's an interesting idea, and would also get rid of some of the other complexity we have with needing a separate port for the ExtendedPackets. Will take a look.